EvalData/models/pairwise_assessment_document.py

```diff
@@ -1041,52 +1787,4 @@ def get_system_data(
 
         return system_data
 
-    @classmethod
-    def get_system_status(cls, campaign_id=None, sort_index=3):
-        system_scores = cls.get_system_scores(campaign_id=None)
-        non_english_codes = (
-            'cs',
-            'de',
-            'fi',
-            'lv',
-            'tr',
-            'tr',
-            'ru',
-            'zh',
-        )
-
-        codes = ['en-{0}'.format(x) for x in non_english_codes] + [
-            '{0}-en'.format(x) for x in non_english_codes
-        ]
-
-        data = {}
-        for code in codes:
-            data[code] = {}
-            for key in [x for x in system_scores if code in x]:
-                data[code][key] = system_scores[key]
-
-        output_data = {}
-        for code in codes:
-            total_annotations = sum([len(x) for x in data[code].values()])
-            output_local = []
-            for key in data[code]:
-                x = data[code][key]
-                z = sum(x) / total_annotations
-                output_local.append((key, len(x), sum(x) / len(x), z))
-
-            output_data[code] = list(
-                sorted(output_local, key=lambda x: x[sort_index], reverse=True)
-            )
-
-        return output_data
-
-    @classmethod
-    def completed_results_for_user_and_campaign(cls, user, campaign):
-        results = cls.objects.filter(
-            activated=False,
-            completed=True,
-            createdBy=user,
-            task__campaign=campaign,
-        ).values_list('item_id', flat=True)
 
-        return len(set(results))

```

Comment:
It's quite difficult to figure out what exactly has happened in this file. I believe all is good, but can you briefly summarize the changes here?

Reply:
Summarize:
1. Introduction of Abstract Base Classes:
Created BasePairwiseAssessmentDocumentTask and BasePairwiseAssessmentDocumentResult abstract base classes to move common fields and methods into these base classes, reducing code duplication.
2. New and Modified Fields:
Added mqm1 and mqm2 fields to the TextSegmentWithTwoTargetsWithContext class for storing MQM tutorial instructions.
3. Method Adjustments:
Moved common methods to the abstract base classes for reuse across multiple subclasses.
4. New Task and Result Classes:
Added PairwiseAssessmentDocumentESATask and PairwiseAssessmentDocumentESAResult classes to handle tasks and results with error span annotations.
5. Export Functionality:
Enhanced the export functionality in the PairwiseAssessmentDocumentESAResult class to include the mqm1 and mqm2 fields.

Reply:
Thanks a lot, that's helpful!